src/GoTrueClient.ts

```diff
@@ -593,6 +595,43 @@ export default class GoTrueClient {
     }
   }
 
+  /**
+   * Returns a new session, from the current session, regardless of expiry status.
+   * If the access token or refresh token are invalid, an error will be thrown.
+   * @param currentSession The current session that minimally contains an access token and refresh token.
+   */
+  async refreshSession(
+    currentSession: Pick<Session, 'access_token' | 'refresh_token'>
+  ): Promise<AuthResponse> {
+    try {
+      if (!currentSession.access_token || !currentSession.refresh_token) {
+        throw new AuthSessionMissingError()
+      }
```

Comment:
To refresh a session you only need a refresh token, not an access token. I don't see why it's being used here. 

Can you make the refresh token optional; if not provided you should look it up using `getSession()`.

Reply:
I was basing this on the `setSession()` method. I can remove it.

Reply:
One comment about making the `refresh_token` optional:

A use-case for this method is to use it in a server-side context, which means there wouldn't be a session returned from `getSession()`. So I'm thinking we can make `currentSession` optional, but if it's passed in, then it must contain a `refresh_token`. I assume `currentSession?: Pick<Session, 'refresh_token'>` would do this??

Am I thinking about this correctly?